### DISCUSSION TITLE
Does vSwarm support Firecracker-Containerd

### DISCUSSION THREAD

**Discussion by DanielLee343** [vhive-serverless/vSwarm#348] [Q&A] [answered]
Hi, I noticed that vSwarm only supports applications deployed and invoked by Knative, is that true? Do we have a feature to run those functions via Firecracker-containerd? It's interesting to see how real-world FaaS applications perform snapshots. Or do you have any advise to walk around this? Thanks!

**Comment by ustiugov** [comment, marked as answer]
@DanielLee343 yes, all vHive runtimes are supported (but only container-based setting is tested in CI). However, to use Firecracker, you need to modify YAML files, which is the limitation we are currently working on. 

More details: https://github.com/vhive-serverless/vHive/blob/main/docs/quickstart_guide.md#iv-deploying-and-invoking-functions-in-vhive

**Reply by jingren1021** [reply]
Aside YAML files, I am assuming function files and tracing, storage and util files need to be modified too since they take environment variables as well. Or is there some way to pass env variables into the guest image?

Is there any working function image example for Firecracker now?

**Reply by ustiugov** [reply]
We are working on merging the fix in vHive for env variables: https://github.com/vhive-serverless/vHive/pull/633

**Reply by ustiugov** [reply]
working images are in the vHive repo and workflows

**Reply by ustiugov** [reply]
@jingren1021

**Reply by jingren1021** [reply]
Thanks for the update!
